### ISSUE TITLE
DB Refresh: 0 members sync issue: TermHub v ConceptSetBrowser

### ISSUE THREAD

**Issue by hlehmann17** [jhu-bids/TermHub#402]
### Overview
Mismatch between TermHub and CS Browser

### To Reproduce
In prod:
1. go to Help/Issues page
2. Past 563193300 into the How to: Load a set of concept sets box
3. Go (back) to CSet Search
4. Load concepts (already says 1 member, 0 conceps)
5. Go to CSet Comparison: No concepts
6. In concept set browser, locate concept set with version id 563193300
7. Click OMOP Concepts. See the listing of 43 concepts ≠ 0 of TermHub
(8. Clicking, in Help/Issues, on Refresh Database, does not seem to do anything)

### What browser are you using?
macOS Chrome

### Expected behavior _(optional section)_
I should see 43 concepts. (I should also see the concept set listed, when I type in Imdevimab into the search box)

### Actual behavior _(optional section)_
As above

### Screenshots _(optional section)_
<>

### Additional context _(optional section)_
Same behavior 2 days ago, when I created the concept set. I thought that, with 2 nights to update itself, I would not see this problem.

## Related
https://github.com/jhu-bids/TermHub/issues/451

**Comment by joeflack4** [comment]
@hlehmann17 So sorry about this! Your guinea piggery, err, testing, is highly appreciated!

## Source of the problem
I know what the problem is. The enclave's `concept_set_members` are not as up-to-date as its other data. When we do the refresh, it's fetching new containers, versions, and expression items, but not the members.

<img width="479" alt="Screen Shot 2023-06-16 at 8 48 28 PM" src="https://github.com/jhu-bids/TermHub/assets/13045020/94ca73ef-87c9-4e96-a046-35be55978307">

I just verified and see that the enclave does show these members are there now. However, the DB refresh thinks that it's already fetched everything needed for this concept set version, so I'll need to fix that.

## Possible solutions
I prefer in the order of: **d** (_in cases where we push from termhub_) > **c** > **b** > **a**
### a. Deny fetching when cset returns >0 expression items, but 0 members
This would leads to some poor UX though. A user would rightfully wonder why they a cset wasn't fetched that they knew to exist. Although if we went this route, we could display an error message about it.
### b. fetch it and update later
This would lead to poor UX though in the interim between when the it first fetches the cset, but doesn't get its version items, and when it fetches again and populates its version items. Although we could ease that a little bit by flagging the cset in some way, like with a warning sign or pink highlight and a tooltip or something.
### c. See if the enclave can get us updated `concept_set_members` faster?
I imagine this probably is not an option. But maybe we could ask the infra team? I don't know what kinds of solutions might be possible here.
### d. Temporarily store unsynced data
**This only works in cases where we push the data.** 
Basically, if we're pushing from TermHub to the enclave, rather than updating our database based on what we fetch from it, we can update our database with data that we would expect to see from the enclave. We could create a table to store records of these "unsynced" objects. We could display something in the UI to indicate that these objects are unsynced. Then, when we fetch the data, we remove that record from the unsynced table.

**Comment by Sigfried** [comment]
@joeflack4 -- You can't accuse someone who keeps kosher of guinea piggery. I'll be surprised if @hlehmann17 ever speaks to you again.  :)

But, seriously...  We should talk more about this. Maybe anytime we have a version with expression terms (aka, items) but no expansion terms (aka, members), if the version created_at is less than an hour ago, we set a flag in the code_sets table (or, probably better, in all_csets) and the UI will report that expansion terms may not be available yet; then try to refetch them every few minutes until we think we've got all of them (could be tricky figuring that out), then remove the flag.

**Comment by joeflack4** [comment]
@Sigfried What you propose is a variation on my proposal **b**; I just put it down as **b<sub>iii</sub>**. However this does require some effort as we'd probably have to set up a worker process.
What do you think of my new proposal **e** above?

**Comment by joeflack4** [comment]
@hlehmann17 I fixed the problem for the concept set you mentioned (as well as any other set in the last 2 weeks or so since refresh launched):
<img width="608" alt="Screen Shot 2023-06-21 at 10 44 08 PM" src="https://github.com/jhu-bids/TermHub/assets/13045020/4b235514-51df-45a8-b09c-7b2fee745291">

For the time being, I've went with option (a). Even though it is the least preferable, it is the only one that I had time for right now.

**Comment by joeflack4** [comment]
Deprioritizing now because we have some fixes in place. Upping the refresh rate as well to 20 minutes. If this continues to come up as a  more majorUX issue, we can increase priority and implement one of the other solutions.

**Comment by joeflack4** [comment]
@Sigfried I updated the issue so that it's clear the implementation that we're going to try to go with.

I added all of the other ideas in a collapsible block.
